### ISSUE TITLE
Derive the name of a graphic from a URL where no alt or title attribute is provided

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#130]
**Reported by jteh on 21 Mar 2008 03:02 UTC**
Unfortunately, there are many web pages which do not provide alt or title attributes for images. Currently, NVDA provides no text for the graphic at all in these cases. For graphics included in links, the name of the graphic should be derived from the URL of either the link or the graphic itself. For graphics which are not part of links, the URL of the graphic is the only option.

The URL should be filtered somewhat in an attempt to strip unnecessary information. I propose that the filename portion of the URL be used with the extension stripped. If a "#" and/or "?" portion is included, this should be maintained, as it may be required to distinguish two links.

== Test Case ==
 * See attachment; includes untagged graphics alone, inside of a link and with an onClick event.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 20 May 2008 00:57 UTC**
Recent versions of the gecko_ia2 virtual buffer back-end include the URL of graphics in the buffer. However, they do not filter the URL as suggested here. I have recently written some C++ code to do this, which Mick will add to the library soon.

Note that until recently, the URL of the image (as opposed to the URL of a link) was not exposed by Gecko. MozillaBug:429659 fixes this, exposing the image URL in the "src" IA2 object attribute.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 29 May 2008 03:24 UTC**
This has now been implemented in virtualBufferLibrary (revision 35 in the main bzr branch). It is available in the [NVDA Miscellaneous Dependencies Package](MiscellaneousDependencies) as of version 2008-05-29-01.
